content/docs/architecture_guide/performance.md

```diff
@@ -0,0 +1,245 @@
++++
+title = "Performance Guide"
+weight = 30
++++
+gVisor is designed to provide a secure, virtualized environment while preserving
+key benefits of containerization, such as small fixed overheads and a dynamic
+resource footprint. For containerized infrastructure, this can provide an "easy
+button" for sandboxing untrusted workloads: there are no changes to the
+fundamental resource model.
+
+gVisor imposes runtime costs over native containers. These costs come in two
+forms: additional cycles and memory usage, which may manifest as increased
+latency, reduced throughput or density, or not at all. In general, these costs
+come from two different sources.
+
+First, the existence of the [Sentry](../) means that additional memory will be
+required, and application system calls must traverse additional layers of
+software. The design emphasizes [security](../security/) and therefore we chose
+to use a language for the Sentry that provides benefits in this domain but may
+not yet offer the raw performance of other choices. Costs imposed by these
+design choices are **structural costs**.
+
+Second, as gVisor is an independent implementation of the system call surface,
+many of the subsystems or specific calls are not as optimized as more mature
+implementations. A good example here is the network stack, which is continuing
+to evolve but does not support all the advanced recovery mechanisms offered by
+other stacks and is less CPU efficient. This an **implementation cost** and is
+distinct from **structural costs**. Improvements here are ongoing and driven by
+the workloads that matter to gVisor users and contributors.
+
+This page provides a guide for understanding baseline performance, and calls out
+distint **structural costs** and **implementation costs**, highlighting where
+improvements are possible and not possible.
+
+While we include a variety of workloads here, it’s worth emphasizing that gVisor
+may not be an appropriate solution for every workload, for reasons other than
+performance. For example, a sandbox is likely to provide minimal benefit for
```

Comment:
There are valid cases for isolating databases, like isolating databases from different users (Wordpress comes to mind). Another example is to reduce the blast radius if the database sandbox gets compromised, isolating it from the sandbox with credit card information running along side it.

Maybe replace it with guidance around testing their own application workload to get realistic numbers for their particular use case?

Reply:
I've clarified this line to say "trusted database". Note that this paragraph is about emphasizing that this isn't a solution that you turn on everywhere and get magic security.